pyxrf/xanes_maps/xanes_maps_api.py

```diff
@@ -2283,6 +2358,30 @@ def btn_stack_prev_clicked(self, event):
             if n_current > 0:
                 self.switch_to_different_stack(self.labels[n_current - 1])
 
+        def btn_save_spectrum_clicked(self, event):
+            # Create file name for the CSV file
+            fln = "spectrum"
+            fln += f"_{self.label_selected}"
+            x_min, y_min, x_max, y_max = self.pts_selected
+            x_str = f"_x_{x_min}" if (x_min == x_max) else f"_x_{x_min}_{x_max}"
+            y_str = f"_y_{y_min}" if (y_min == y_max) else f"_y_{y_min}_{y_max}"
+            fln += x_str + y_str + ".csv"
```

Comment:
Or ultimately, it can be 
```suggestion
            fln = f"spectrum_{self.label_selected}{x_str}{y_str}.csv"
```